src/sbf.cpp

```diff
@@ -1,590 +1,713 @@
+		_gps_position->vel_d_m_s = -1.0f * static_cast<float>(_buf.payload_pvt_geodetic.vu);
+		_gps_position->vel_m_s = sqrtf(_gps_position->vel_n_m_s * _gps_position->vel_n_m_s +
+					       _gps_position->vel_e_m_s * _gps_position->vel_e_m_s);
+
+		_gps_position->cog_rad = static_cast<float>(_buf.payload_pvt_geodetic.cog) * M_DEG_TO_RAD_F;
+		_gps_position->c_variance_rad = 1.0f * M_DEG_TO_RAD_F;
+
+		// _buf.payload_pvt_geodetic.cog is set to -2*10^10 for velocities below 0.1m/s
+		if (_buf.payload_pvt_geodetic.cog > 360.0f) {
+			_buf.payload_pvt_geodetic.cog = NAN;
+		}
+
+		_gps_position->time_utc_usec = 0;
+#ifndef NO_MKTIME
+		/* convert to unix timestamp */
+		memset(&timeinfo, 0, sizeof(timeinfo));
+
+		timeinfo.tm_year = 1980 - 1900;
+		timeinfo.tm_mon = 0;
+		timeinfo.tm_mday = 6 + _buf.WNc * 7;
+		timeinfo.tm_hour = 0;
+		timeinfo.tm_min = 0;
+		timeinfo.tm_sec = _buf.TOW / 1000;
+
+		epoch = mktime(&timeinfo);
+
+		if (epoch > GPS_EPOCH_SECS) {
+			// FMUv2+ boards have a hardware RTC, but GPS helps us to configure it
+			// and control its drift. Since we rely on the HRT for our monotonic
+			// clock, updating it from time to time is safe.
+
+			timespec ts;
+			memset(&ts, 0, sizeof(ts));
+			ts.tv_sec = epoch;
+			ts.tv_nsec = (_buf.TOW % 1000) * 1000 * 1000;
+			setClock(ts);
+
+			_gps_position->time_utc_usec = static_cast<uint64_t>(epoch) * 1000000ULL;
+			_gps_position->time_utc_usec += (_buf.TOW % 1000) * 1000;
+		}
+
+#endif
+		_gps_position->timestamp = gps_absolute_time();
+		_last_timestamp_time = _gps_position->timestamp;
+		_rate_count_vel++;
+		_rate_count_lat_lon++;
+		ret |= (_msg_status == 7) ? 1 : 0;
+		//SBF_DEBUG("PVTGeodetic handled");
+		break;
+
+	case SBF_ID_VelCovGeodetic: SBF_TRACE_RXMSG("Rx VelCovGeodetic");
+		_msg_status |= 2;
+		_gps_position->s_variance_m_s = _buf.payload_vel_col_geodetic.cov_ve_ve;
+
+		if (_gps_position->s_variance_m_s < _buf.payload_vel_col_geodetic.cov_vn_vn) {
+			_gps_position->s_variance_m_s = _buf.payload_vel_col_geodetic.cov_vn_vn;
+		}
+
+		if (_gps_position->s_variance_m_s < _buf.payload_vel_col_geodetic.cov_vu_vu) {
+			_gps_position->s_variance_m_s = _buf.payload_vel_col_geodetic.cov_vu_vu;
+		}
+
+		//SBF_DEBUG("VelCovGeodetic handled");
+		break;
+
+	case SBF_ID_DOP: SBF_TRACE_RXMSG("Rx DOP");
+		_msg_status |= 4;
+		_gps_position->hdop = _buf.payload_dop.hDOP * 0.01f;
+		_gps_position->vdop = _buf.payload_dop.vDOP * 0.01f;
+		//SBF_DEBUG("DOP handled");
+		break;
+
+	case SBF_ID_AttEuler: SBF_TRACE_RXMSG("Rx AttEuler");
+
+		if (!_buf.payload_att_euler.error_not_requested) {
+
+			int error_aux1 = _buf.payload_att_euler.error_aux1;
+			int error_aux2 = _buf.payload_att_euler.error_aux2;
+
+			// SBF_DEBUG("Mode: %u", _buf.payload_att_euler.mode)
+			if (error_aux1 == 0 && error_aux2 == 0) {
+				float heading = _buf.payload_att_euler.heading;
+				heading *= M_PI_F / 180.0f; // deg to rad, now in range [0, 2pi]
+
+				// heading -= _heading_offset; // range: [-pi, 3pi]

```

Comment:
Why is this commented?

Reply:
Because the heading offsets are calculated inside the receiver (so no need for that calculation), I send the setAttitudeOffset cmd at configuration to the receiver (line 175)

Reply:
Ok then better remove it here.

Reply:
```suggestion
```